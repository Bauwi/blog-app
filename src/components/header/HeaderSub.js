import React from 'react';
import { NavLink } from 'react-router-dom';

const HeaderSub = () => (
  <nav className="header__navbarsub">
    <div className="header-navbarsub--three">
      <NavLink to="/category/music" activeClassName="header__navbarsub__tab--selected">
        Music
      </NavLink>
      <NavLink to="/category/movie" activeClassName="header__navbarsub__tab--selected">
        Movie
      </NavLink>
      <NavLink to="/category/litterature" activeClassName="header__navbarsub__tab--selected">
        Litterature
      </NavLink>
    </div>
    <div className="header-navbarsub--three">
      <NavLink to="/category/life" activeClassName="header__navbarsub__tab--selected">
        Life
      </NavLink>
      <NavLink to="/category/programming" activeClassName="header__navbarsub__tab--selected">
        Programming
      </NavLink>
      <NavLink to="/category/history" activeClassName="header__navbarsub__tab--selected">
        History
      </NavLink>
    </div>
    <div className="header-navbarsub--three">
      <NavLink to="/category/sport" activeClassName="header__navbarsub__tab--selected">
        Sport
      </NavLink>
      <NavLink to="/category/business" activeClassName="header__navbarsub__tab--selected">
        Business
      </NavLink>
    </div>
  </nav>
);

export default HeaderSub;
