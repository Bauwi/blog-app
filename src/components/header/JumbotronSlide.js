import React, { Component } from 'react';
import moment from 'moment';

export class JumbotronSlide extends Component {
  render() {
    const category = this.props.post.keywords.split(',')[0].trim();
    return (
      <div className="carousel-item">
        <div className="carousel-item__image">
          <img src={this.props.post.miniCover} alt="suggestion cover" />
        </div>
        <div className="carousel-item__infos">
          <header>
            <div>
              <p>
                <i className="fa fa-star" /> {this.props.post.stars}
              </p>
              <p className={`category border-${category}`}>{category}</p>
            </div>
            <p>{moment(this.props.post.createdAt).format('MMM Do, YYYY')}</p>
          </header>
          <h2 className="carousel-item_title">{this.props.post.title}</h2>
          <footer>
            <p>by {this.props.post.author}</p>
          </footer>
        </div>
      </div>
    );
  }
}

export default JumbotronSlide;
