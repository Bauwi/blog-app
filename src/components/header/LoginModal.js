import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Modal } from 'antd';
import { startLogin } from '../../actions/auth';

export class LoginModal extends Component {
  state = {
    visible: false,
    confirmLoading: false
  };
  showModal = () => {
    this.setState({
      visible: true
    });
  };

  handleCancel = () => {
    this.setState({
      visible: false
    });
  };
  render() {
    const { visible, confirmLoading } = this.state;
    return (
      <div className="login__modal">
        <button className="button button--navbar" onClick={this.showModal}>
          <i className="fa fa-sign-in " />
        </button>
        <Modal
          title="Bloyster"
          visible={visible}
          confirmLoading={confirmLoading}
          onCancel={this.handleCancel}
          footer={[]}
        >
          <div className="login__container">
            <h1 className="login__title">Share !</h1>
            <button className="button" onClick={this.props.startLoginWithGoogle}>
              <i className="fa fa-google" /> Login With Google
            </button>
            <button className="button" onClick={this.props.startLoginWithGithub}>
              <i className="fa fa-github" /> Login With Github
            </button>
          </div>
        </Modal>
      </div>
    );
  }
}

LoginModal.propTypes = {
  startLoginWithGoogle: PropTypes.func.isRequired,
  startLoginWithGithub: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => ({
  startLoginWithGoogle: () => dispatch(startLogin('google')),
  startLoginWithGithub: () => dispatch(startLogin('github'))
});

export default connect(undefined, mapDispatchToProps)(LoginModal);
