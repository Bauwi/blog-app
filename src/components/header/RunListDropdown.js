import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import { Menu, Dropdown, Icon, message } from 'antd';

import { setCurrentPostRun } from '../../actions/run';

const warning = () => {
  message.warning('You have already read all the posts contained in your run.');
};

export class RunListDropdown extends Component {
  renderDropdownPosts = () => {
    return this.props.posts.map(post => {
      return (
        <Menu.Item key={post.content.id}>
          <div
            className="dropdown-line"
            onClick={() => {
              this.props.setCurrentPostRun(post.content.id);
              this.props.history.push('/run/start');
            }}
          >
            <img src={post.content.miniCover} alt="minicover" />
            <p>{post.content.title}</p>
          </div>
        </Menu.Item>
      );
    });
  };

  handleResumeClick = () => {
    const firstUnreadIndex = this.props.posts.findIndex(post => post.state === 'unread');
    if (firstUnreadIndex === -1) {
      warning();
      this.props.setCurrentPostRun(this.props.posts[0].content.id);
      return this.props.history.push('/run');
    }
    this.props.setCurrentPostRun(this.props.posts[firstUnreadIndex].content.id);
    return this.props.history.push('/run/start');
  };

  render() {
    console.log(this.props);
    const menu = (
      <Menu>
        <Menu.Item key="0">
          <button className="button" onClick={this.handleResumeClick}>
            Resume
          </button>
          <button className="button">Reset</button>
        </Menu.Item>
        {this.props.posts && this.renderDropdownPosts()}
      </Menu>
    );
    return (
      <Dropdown overlay={menu} placement="bottomCenter" className="dropdown">
        <Link className="ant-dropdown-link" to="/run">
          <i className="fa fa-book" />
        </Link>
      </Dropdown>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  setCurrentPostRun: id => dispatch(setCurrentPostRun(id))
});

const mapStateToProps = state => ({
  posts: state.run.posts
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(RunListDropdown));
