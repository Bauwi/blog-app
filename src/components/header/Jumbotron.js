import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import PublicPostList from '../PublicPostsList';
import { Carousel } from 'antd';

import JumbotronSlide from './JumbotronSlide';

import selectCarouselPosts from '../../selectors/carousel';

export class Jumbotron extends Component {
  state = {
    currentSlide: 1
  };
  renderSlides = slideIndex => {
    const post = this.props.posts[slideIndex];
    return <JumbotronSlide post={post} />;
  };
  handleClick = () => {
    const pathname = `${this.props.posts[this.state.currentSlide - 1].authorId}/read/${this.props
      .posts[this.state.currentSlide - 1].id}`;
    this.props.history.push(pathname);
  };
  render() {
    if (this.props.isLoading || this.props.posts.length === 0) {
      return <p />;
    }
    return (
      <div className="carousel">
        <Carousel
          effect="fade"
          autoplay
          autoplaySpeed={7000}
          afterChange={currentSlide => {
            if (this.state.currentSlide === 6) {
              return this.setState(() => ({ currentSlide: 1 }));
            }
            this.setState({ currentSlide: currentSlide + 1 });
          }}
        >
          <div onClick={this.handleClick}>{this.renderSlides(0)}</div>
          <div onClick={this.handleClick}>{this.renderSlides(1)}</div>
          <div onClick={this.handleClick}>{this.renderSlides(2)}</div>
          <div onClick={this.handleClick}>{this.renderSlides(3)}</div>
          <div onClick={this.handleClick}>{this.renderSlides(4)}</div>
          <div onClick={this.handleClick}>{this.renderSlides(5)}</div>
        </Carousel>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  posts: selectCarouselPosts(
    state.readings.posts,
    !!state.users.preferences
      ? state.users.preferences.topCategories
      : ['music', 'literature', 'cinema']
  ),
  isLoading: state.readings.isLoading
});

export default withRouter(connect(mapStateToProps)(Jumbotron));
