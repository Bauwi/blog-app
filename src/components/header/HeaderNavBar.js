import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { Badge } from 'antd';

import RunListDropdown from './RunListDropdown';
import LoginModal from './LoginModal';

import { startLogout } from '../../actions/auth';
import { startSetRunPosts } from '../../actions/run';

export class HeaderNavBar extends Component {
  componentDidMount() {
    if (this.props.isAuthenticated) {
      this.props.startSetRunPosts();
    }
  }

  render() {
    const { isAuthenticated, runCount = 0, startLogout } = this.props;
    if (isAuthenticated) {
      return (
        <nav className="header__navbar">
          <NavLink to="/">
            <i className="fa fa-home" />
          </NavLink>
          <NavLink to="/dashboard">
            <i className="fa fa-pencil" />
          </NavLink>
          <nav to="/run">
            <Badge className="header__navbar__run__badge" showZero count={runCount} />
            <RunListDropdown />
          </nav>
          <NavLink to="/preferences">
            <i className="fa fa-cog" />
          </NavLink>
          <button className="button button--navbar" onClick={startLogout}>
            <i className="fa fa-sign-out" />
          </button>
        </nav>
      );
    }
    return (
      <nav className="header__navbar">
        <NavLink to="/">
          <i className="fa fa-home" />
        </NavLink>
        <LoginModal />
      </nav>
    );
  }
}

HeaderNavBar.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  runCount: PropTypes.number,
  startLogout: PropTypes.func.isRequired,
  startSetRunPosts: PropTypes.func.isRequired
};

HeaderNavBar.defaultProps = { runCount: 0 };

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
  startSetRunPosts: () => dispatch(startSetRunPosts())
});

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid,
  runCount: state.run.posts ? state.run.posts.length : 0
});

export default connect(mapStateToProps, mapDispatchToProps)(HeaderNavBar);
