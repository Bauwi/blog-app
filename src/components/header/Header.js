import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Headroom from 'react-headroom';

import HeaderNavBar from './HeaderNavBar';

const Header = () => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/">
          <h1>_TEKKNE_</h1>
        </Link>
        <HeaderNavBar />
      </div>
    </div>
  </header>
);

export default Header;
