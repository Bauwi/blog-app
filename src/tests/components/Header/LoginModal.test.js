import React from 'react';
import { shallow } from 'enzyme';
import { LoginModal } from '../../../components/header/LoginModal';

test('should render LoginPage correctly', () => {
  const wrapper = shallow(<LoginModal startLoginWithGoogle={() => {}} startLoginWithGithub={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('should hide modal on mount', () => {
  const wrapper = shallow(<LoginModal startLoginWithGoogle={() => {}} startLoginWithGithub={() => {}} />);

  expect(wrapper.state().visible).toEqual(false);
});

test('should open modal on navbar button click', () => {
  const wrapper = shallow(<LoginModal startLoginWithGoogle={() => {}} startLoginWithGithub={() => {}} />);
  wrapper
    .find('button')
    .at(0)
    .simulate('click');

  expect(wrapper.state().visible).toEqual(true);
});

test('should call startLoginGoogle on Login With Google click', () => {
  const startLoginWithGoogle = jest.fn();
  const startLoginWithGithub = jest.fn();
  const wrapper = shallow(<LoginModal
    startLoginWithGoogle={startLoginWithGoogle}
    startLoginWithGithub={startLoginWithGithub}
  />);
  wrapper
    .find('button')
    .at(1)
    .simulate('click');

  expect(startLoginWithGithub).not.toHaveBeenCalled();
  expect(startLoginWithGoogle).toHaveBeenCalled();
});

test('should call startLoginGithub on Login With Github click', () => {
  const startLoginWithGoogle = jest.fn();
  const startLoginWithGithub = jest.fn();
  const wrapper = shallow(<LoginModal
    startLoginWithGoogle={startLoginWithGoogle}
    startLoginWithGithub={startLoginWithGithub}
  />);
  wrapper
    .find('button')
    .at(2)
    .simulate('click');
  expect(startLoginWithGoogle).not.toHaveBeenCalled();
  expect(startLoginWithGithub).toHaveBeenCalled();
});
