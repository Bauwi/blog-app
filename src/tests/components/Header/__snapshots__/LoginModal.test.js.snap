// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="login__modal"
>
  <button
    className="button button--navbar"
    onClick={[Function]}
  >
    <i
      className="fa fa-sign-in "
    />
  </button>
  <Modal
    confirmLoading={false}
    footer={Array []}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    prefixCls="ant-modal"
    title="Bloyster"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <div
      className="login__container"
    >
      <h1
        className="login__title"
      >
        Share !
      </h1>
      <button
        className="button"
        onClick={[Function]}
      >
        <i
          className="fa fa-google"
        />
         Login With Google
      </button>
      <button
        className="button"
        onClick={[Function]}
      >
        <i
          className="fa fa-github"
        />
         Login With Github
      </button>
    </div>
  </Modal>
</div>
`;
